pydantic_ai_slim/pydantic_ai/messages.py

```diff
@@ -176,29 +173,31 @@ def has_content(self) -> bool:
             return bool(self.args.args_json)
 
 
+ModelResponsePart = Annotated[Union[TextPart, ToolCallPart], pydantic.Discriminator('kind')]
+
+
 @dataclass
-class ModelStructuredResponse:
-    """A structured response from a model.
+class ModelResponse:
+    """A response from a model."""
 
-    This is used either to call a tool or to return a structured response from an agent run.
-    """
+    parts: list[ModelResponsePart]
+    """The parts of the response."""
 
-    calls: list[ToolCall]
-    """The tool calls being made."""
+    role: Literal['model-response'] = 'model-response'
```

Comment:
This would be breaking, but I think it would be valuable to just have `kind` as `'model-response'` and `role` as `'assistant'`. Then we could do the same thing for these other message types, with `role` either being `'user'` or `'assistant'` (like Anthropic), and `kind` being the discriminator that's helpful for our use cases.

Reply:
Discussed with @dmontagu offline, I'll do this in a separate PR before we do our upcoming release :)